multi_physics/QED/kokkos_example/kokkos_example_breit_wheeler.cpp

```diff
@@ -0,0 +1,342 @@
+* @param[in] chi_min the minimum chi parameter
+* @param[in] chi_max the maximum chi parameter
+* @param[in] chi_size the size of the lookup table along the chi axis
+* @return the lookup table
+*/
+template <typename Real, typename Vector>
+auto generate_dndt_table(Real chi_min, Real chi_max, int chi_size)
+{
+    std::cout << "Preparing dndt table [" << get_type_name<Real>()
+        << ", " << chi_size <<"]...\n";
+    std::cout.flush();
+
+    pxr_bw::dndt_lookup_table_params<Real> bw_params{chi_min, chi_max, chi_size};
+
+	auto table = pxr_bw::dndt_lookup_table<
+        Real, Vector>{bw_params};
+
+    table.generate();
+
+    return table;
+}
+
+/**
+* Generates the pair production lookup table
+*
+* @tparam Real the floating point type to be used
+* @tparam Vector the vector type to be used
+* @param[in] chi_min the minimum chi parameter
+* @param[in] chi_max the maximum chi parameter
+* @param[in] chi_size the size of the lookup table along the chi axis
+* @param[in] frac_size the size of the lookup table along the frac axis
+* @return the lookup table
+*/
+template <typename Real, typename Vector>
+auto generate_pair_table(Real chi_min, Real chi_max, int chi_size, int frac_size)
+{
+    std::cout << "Preparing pair production table [" << get_type_name<Real>()
+        << ", " << chi_size << " x " << frac_size <<"]...\n";
+    std::cout.flush();
+
+    pxr_bw::pair_prod_lookup_table_params<Real> bw_params{
+        chi_min, chi_max, chi_size, frac_size};
+
+	auto table = pxr_bw::pair_prod_lookup_table<
+        Real, Vector>{bw_params};
+
+    table.template generate();
+
+    return table;
+}
+
+/**
+* Tests the initialization of the optical depth
+*
+* @tparam Real the floating point type to be used
+* @tparam TableType the lookup table type
+* @param[in,out] pdata the particle data
+* @param[in] repetitions how many times should the test be repeated
+* @param[in,out] rand_pool a random pool
+* @return a bool success flag and the elapsed time in ms, packed in a pair
+*/
+template <typename Real>
+std::pair<bool, double>
+fill_opt_test(
+    ParticleData<Real>& pdata,
+    const int repetitions,
+    Kokkos::Random_XorShift64_Pool<>& rand_pool)
+{
+    Kokkos::Timer timer;
+    for(int rr = 0; rr < repetitions; ++rr){
+        const auto num_particles = pdata.num_particles;
+        Kokkos::parallel_for("FillOpt_"+get_type_name<Real>(),
+            num_particles, KOKKOS_LAMBDA(int i){
+            auto rand_gen = rand_pool.get_state();
+            pdata.m_fields.opt(i) =
+                pxr_bw::get_optical_depth<Real>(
+                    get_rand<Real>::get(rand_gen));
+            rand_pool.free_state(rand_gen);
+        });
+    }
+
+    Kokkos::fence();
+    const auto time = timer.seconds()*1000.0;
+
+    return std::make_pair(check(pdata.m_fields.opt, true, false), time);
+}
+
+/**
+* Tests the evolution of the optical depth
+*
+* @tparam Real the floating point type to be used
+* @param[in,out] pdata the particle data
```

Comment:
```suggestion
* @tparam Real the floating point type to be used
* @tparam TableType the lookup table type
* @param[in,out] pdata the particle data
```

But here we have the TableType.